robyn/router.py

```diff
@@ -124,11 +124,44 @@ def add_route(
         exception_handler: Optional[Callable],
         injected_dependencies: dict,
     ) -> Union[Callable, CoroutineType]:
+        def wrapped_handler(*args, **kwargs):
+            # In the execute functions the request is passed into *args
+            requests = list(filter(lambda a: isinstance(a, Request), args))
+
+            handler_params = signature(handler).parameters
+
+            if not requests or (len(handler_params) == 1 and next(iter(handler_params)) is Request):
+                return handler(*args, **kwargs)
+
+            request = requests[0]
+
+            request_components = {
+                "r": request,
+                "request": request,
+                "query_params": request.query_params,
+                "headers": request.headers,
+                "path_params": request.path_params,
+                "body": request.body,
+                "method": request.method,
+                "url": request.url,
+                "ip_addr": request.ip_addr,
+                "identity": request.identity,
+                "form_data": request.form_data,
+                "files": request.files,
+                "router_dependencies": injected_dependencies["router_dependencies"],
+                "global_dependencies": injected_dependencies["global_dependencies"],
+                **kwargs,
+            }
```

Comment:
This is great! 

But I want you to handle two more edge cases.

If types are present, then I want you to give the types to the priority.

If there are no types and there is only one parameter, we should assign it to `request` ? What say?

and finally - should we also add `req`: request?

Reply:
@sansyrox 

> If types are present, then I want you to give the types to the priority.

this would only work for query params, headers, url and identity, right? how would it work for params with generic types, say a dict or a string?

    query_params: QueryParams
    headers: Headers
    path_params: dict[str, str]
    body: Union[str, bytes]
    method: str
    url: Url
    form_data: dict[str, str]
    files: dict[str, bytes]
    ip_addr: Optional[str]
    identity: Optional[Identity]

> If there are no types and there is only one parameter, we should assign it to `request` ? What say?

this doesn't make sense. what if we have `def fun(query_params):`?

> and finally - should we also add `req`: request?

alright.

Reply:
> this doesn't make sense. what if we have def fun(query_params):?

fair

> this would only work for query params, headers, url and identity, right? how would it work for params with generic types, say a dict or a string?

we should implement custom types here maybe using type alias